### PR TITLE
Stop auto-extending TTL on save/create paths (closes #233)

### LINKED ISSUE #233: Restrict TTL setting to first creation only — skip refresh from asave and other creation paths
## Summary

Currently `refresh_ttl_if_needed` is invoked from creation/save paths (e.g. `asave`, `aduplicate`, `aduplicate_many`, `ainsert`) with `initial=True`. When the action also belongs to a refresh group (`ActionGroup.UPDATE`/`ActionGroup.CREATE`), the TTL is set/extended every time these methods run — even when the model already exists and TTL was set on an earlier creation. We want TTL to be set **only at the very first creation** of the key and not be touched again by subsequent `asave`/creation calls.

## Features / Details

- TTL on the main key (and special-field keys) must only be assigned during the first creation of the Redis record.
- Calling `asave` on an already-existing model must not extend or reset the TTL, regardless of whether the action is registered in the refresh actions.
- The same rule applies to other creation/update entry points that currently flow through `refresh_ttl_if_needed` with `initial=True`: `aduplicate`, `aduplicate_many`, `ainsert`, etc.
- Continue to honor the explicit refresh path: scheduled/manual TTL refresh and `aset_ttl` should still work as today.
- `nx=True` semantics on `EXPIRE` must be enforced from the call site so we never overwrite an existing TTL during a save/update.

## Use Cases

- A model is saved once with a TTL, then later updated repeatedly via `asave`. The TTL should keep counting down from the original creation, not be reset on every save.
- A pipeline that calls `asave` after mutating fields should not silently extend the lifetime of the key.
- Creation-style helpers like `aduplicate`/`aduplicate_many` should leave TTL alone for any pre-existing target key and only initialize TTL on the brand-new duplicates.

## Implementation Considerations

- `rapyer/base.py::refresh_ttl_if_needed` already supports an `nx` flag; the bug surfaces because `asave` is decorated with `@mark_actions(ActionGroup.UPDATE, ActionGroup.CREATE, initial=True, ...)`, so when the model is registered for refresh-on-update, `should_refresh` becomes True and `nx` becomes False — overwriting an existing TTL.
- Decide whether `asave` should drop `initial=True`, or whether the decorator should force `nx=True` for save/creation paths even when `should_refresh` is True.
- Audit every call that passes `initial=True` and confirm none of them should ever overwrite an existing TTL.
- Add tests covering: save → wait → save again → TTL must not be extended; duplicate of an existing model preserves the original's TTL countdown; `aset_ttl` still overwrites as expected.

## List tasks
- [ ] Audit all call sites passing `initial=True` to `refresh_ttl_if_needed` (`asave`, `aduplicate`, `aduplicate_many`, `ainsert`, and any others)
- [ ] Change creation/save paths so TTL is set with `EXPIRE NX` (only when no TTL exists), independent of whether the action is in the refresh group
- [ ] Ensure `asave` on an existing record never resets or extends TTL on the main key or special-field keys
- [ ] Ensure `aduplicate`/`aduplicate_many` only initialize TTL on the freshly created duplicates, not on the source
- [ ] Keep `aset_ttl` and the explicit refresh path behaving as today (overwrite allowed)
- [ ] Add unit tests: repeated `asave` does not extend TTL
- [ ] Add unit tests: `aduplicate`/`aduplicate_many` initialize TTL only on new keys
- [ ] Add integration test verifying TTL countdown survives multiple `asave` calls
- [ ] Update CHANGELOG with the behavior change

### PR DESCRIPTION
**Pull request by yedidyakfir**
## Summary

Save/create paths (`asave`, `aduplicate`, `aduplicate_many`, `ainsert`) no longer extend an existing key's TTL. The `initial=True` mechanism that set TTL implicitly on creation paths is removed; users now opt in explicitly via `Meta.refresh_ttl=ActionGroup.CREATE`.

Also introduces `mark_actions(version="v2")`: the wrap/no-wrap decision for TTL refresh is moved from per-call runtime to model class install time, removing the per-call `should_refresh_for_action` check.

Closes #233

## Changes

- `refresh_ttl_if_needed`: drop the `initial`/`nx` branch; only refreshes when the meta opts in for the action.
- New `refresh_ttl` (unconditional) for the v2 flush path that has already proven the wrap decision at install time.
- `mark_actions(initial=...)` removed from the public API; `asave`, `aduplicate`, `aduplicate_many`, `ainsert` lose their implicit TTL set.
- Per-version flush strategy: `FLUSH_V1` (OR-merges action groups, calls `refresh_ttl_if_needed`) and `FLUSH_V2` (first-registration wins, calls `refresh_ttl` unconditionally).
- v2 install-time wrapping: `install_action_for_meta` wraps only when `should_refresh_for_action(meta, combined)`; no wrap, no per-call cost.
- Tests: deleted `test_initial.py`; `assert_action` helper skips the `action` assertion under v2 (action is consumed at install time, not propagated to refresh). Integration `installed_refresh_ttl` reinstalls the model class so v2's install-time decision actually takes effect.
- README: outdated `set` / lock examples replaced with current API.

## Testing

- `tests/unit/mark_actions/` — updated unit tests cover both v1 and v2 paths via `mark_version` parametrization.
- `tests/integration/actions/ttl.py` — refresh / no-refresh integration tests now reinstall the model so v2 flush is exercised.
- Repeated `asave` no longer extends TTL (covered by the no-refresh integration tests on save paths).

## Summary by CodeRabbit

* **New Features**
  * Exposed JSON client/pipeline accessors and fine-grained TTL helpers; added an is-inner-model helper.

* **Refactor**
  * Reworked action/TTL registration and flush flow with nested-call tracking and two flush strategies; standardized JSON pipeline usage across types.

* **Tests**
  * Updated unit/integration tests, fixtures, and benchmarks to reflect new TTL/install semantics and calling conventions.

* **Documentation**
  * README: added atomic update example and improved lock-context multi-field atomic update example.

### CODE REVIEW

File: rapyer/types/lst.py [hunk 1]
```diff
@@ -158,7 +154,7 @@ async def aextend(self, __iterable):
     async def apop(self, index=-1):
         if self:
             self.pop(index)
-        arrpop = await self.redis.json().arrpop(self.key, self.json_path, index)  # type: ignore[misc]
+        arrpop = await self.Meta.redis_json.arrpop(self.key, self.json_path, index)  # type: ignore[misc]
```
[the remark this answers is not in the document]
Comment (yedidyakfir) [reply rapyer/types/lst.py:158]: like we said in the past, read methods cant use pipeline

File: rapyer/base.py [hunk 2]
```diff
@@ -517,7 +530,7 @@ def _resolve_key(cls, key: str | Self) -> str:
     )
     async def aget(cls, key: str) -> Self:
         key = cls._resolve_key(key)
-        model_dump = await cls.Meta.redis.json().get(key, "$")  # type: ignore[misc]
+        model_dump = await cls.Meta.redis_json.get(key, "$")  # type: ignore[misc]
```
[the remark this answers is not in the document]
Comment (yedidyakfir) [reply rapyer/base.py:533]: same as previous response
